templates/strings/objc-m.stencil

```diff
@@ -0,0 +1,80 @@
+// Generated using SwiftGen — https://github.com/SwiftGen/SwiftGen
+
+{% if tables.count > 0 %}
+
+#import "Localizable.h"
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-security"
+
+@interface BundleToken : NSObject
+@end
+
+@implementation BundleToken
+@end
+
+NSString* tr(NSString* tableName, NSString* key, ...) {
+    va_list args;
+    va_start(args, key);
+
+    NSBundle *bundle = [NSBundle bundleForClass:BundleToken.class];
+    NSString *format = [bundle localizedStringForKey:key value:@"" table:tableName];
+    NSString *result = [[NSString alloc] initWithFormat:format arguments:args];
+    return result;
+};
+
+{% macro parametersBlock types %}{% filter removeNewlines:"leading" %}
+  {% for type in types %}
+    ({% call paramTranslate type %})p{{ forloop.counter }}{% if not forloop.last %} and:{% endif %}
+  {% endfor %}
+{% endfilter %}{% endmacro %}
+{% macro argumentsBlock types %}{% filter removeNewlines:"leading" %}
+  {% for type in types %}
+    p{{forloop.counter}}{% if not forloop.last %}, {% endif %}
+  {% endfor %}
+{% endfilter %}{% endmacro %}
+{% macro paramTranslate swiftType %}
+  {% if swiftType == "Float" %}
+    float
+  {% elif swiftType == "Int" %}
+    NSInteger
+  {% elif swiftType == "CChar" %}
+    char
+  {% elif swiftType == "String" %}
+    NSString*
+  {% elif swiftType == "UnsafePointer<CChar>" %}
+    char*
+  {% elif swiftType == "UnsafeRawPointer" %}
+    void*
```

Comment:
`id` maybe?

Reply:
I don't think so!  This is "%p", which may not point to an ObjC object.

Reply:
Ah now I know what I was missing. There's actually an issue with the generated context (in SwiftGenKit): it parses `%@` as `.object` (good), which is converted for the context to `String` (bad). I'll need to fix that first.

Reply:
See #620 for the necessary change. So for `String` you should generate `id`.

With SwiftGen 7.0 we'll correct the context type to something more generic (like `Object`), but for now check against `String`.

Reply:
I agree with your #620.  I'll change my templates so that when they see either 'String' or 'Any', they emit 'id'.

For the 17th time, I regret not creating an 'objcTypes' next to 'types' (see `./Documentation/SwiftGenKit Contexts/Strings.md#35`)

Reply:
okay, just pushed that commit.

Reply:
Nah, we would never have added a separate `objcTypes` to the context. We're going to make the content of `types` more generic in some future version, so both the obj-c and swift templates will have to convert them to more concrete types.

Reply:
Side note: it could be cool to have a nice Stencil tag or filter for such mappings, like `{{ type | map "String:NSString,Int:NSInteger,Any:id" }}` or something similar to what we'd do with a `switch` or `Dictionary` if it were in Swift

No rush and shouldn't block that PR, but would definitively make things easier in the future for such conversions rather than having to go the `{% if type == 'String' %}NSString{% …` route :wink: